### PR TITLE
Shaul spotify

### PR DESCRIPTION
**Pull request by orelzion**

### CODE REVIEW

File: app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt [hunk 1]
```diff
@@ -3,32 +3,63 @@ package com.github.orelzion.spotifyclone.view
...
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        albumsListView.adapter = albumsAdapter
-        albumsListView.layoutManager = LinearLayoutManager(requireContext())
+        /**
+         * Set albumAdapter's clickListener to AlbumListFragment's onItemClicked method
+         */
+        albumsListView.adapter = albumsAdapter.apply {
```
Comment (orelzion) [inline app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt:32]: Nice! I loved the use of apply. But not sure that it's relevant here. I mean this code is totally acceptable without the apply at all.
```
albumsAdapter.clickListener = this::onItemClicked
```

File: app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt [hunk 2]
```diff
@@ -3,32 +3,63 @@ package com.github.orelzion.spotifyclone.view
...
+
+    /**
+     * onItemClicked - show corresponding tracks' list in main fragment
+     */
+    private fun onItemClicked(albumId: String) {
+        showFragment(R.id.mainFragmentContainer, TracksListFragment.newInstance(albumId), true)
+    }
+
+    // TODO: Ask Orel - is this method should be declared independently,
+    //  so it stays available for both AlbumListFragment and TracksListFragment?
```
Comment (orelzion) [inline app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt:55]: So theoretically yes, we could have written a shared function here, but we would need to replace it outside of this fragment in order for that to be shared.
One way will be to have a `BaseFragment` class with this function and have them both inherit from it, but I find it too limiting.
Another option will be to use extension functions, which is something we didn't learn yet

File: app/src/main/java/com/github/orelzion/spotifyclone/view/TracksListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    companion object {
+        /**
+         * factory method
+         * accessed via TracksListFragment, as it is being declared in companion object
+         */
+        fun newInstance(albumId: String): TracksListFragment {
+            return TracksListFragment().apply {
+                arguments = Bundle().apply {
+                    putString("albumId", albumId)
+                    // TODO: Ask Orel - Shall I declare albumId as a Pair? in/outside companion object?
```
Comment (orelzion) [inline app/src/main/java/com/github/orelzion/spotifyclone/view/TracksListFragment.kt:37]: I'm not sure I get why we need a pair here

**Review by orelzion — COMMENTED** [review]
That was great! I think you really grasped the understanding of how fragments work, and I really enjoyed seeing you playing with RecyclerView and ConstraintLayout so efficiently

File: app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt (hunk 2 above)
Comment (chiefchiefon) [reply app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt:55]: Nice :)
I'll wait for the extentions..

File: app/src/main/java/com/github/orelzion/spotifyclone/view/TracksListFragment.kt (hunk 3 above)
Comment (chiefchiefon) [reply app/src/main/java/com/github/orelzion/spotifyclone/view/TracksListFragment.kt:37]: Pair for key and value

**Comment by chiefchiefon** [comment]
So after reviewing your comments, I'll just hit "Merge pull request"?

**Comment by orelzion** [comment]
> So after reviewing your comments, I'll just hit "Merge pull request"?

Well, usually yes, but not now. Because Bnaya has to do that as well, we will just close it

**Pull request closed without merging**